src/main/kotlin/domain/Cell.kt

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import domain.geometric.Location
+
+sealed class Cell {
+    abstract val location: Location
+
+    data class Mine(override val location: Location) : Cell()
+    data class Safe(override val location: Location) : Cell()
+
+    companion object {
+        fun safe(location: Location): Cell {
+            return Safe(location)
+        }
+
+        fun mine(location: Location): Cell {
+            return Mine(location)
+        }
+
+        fun of(location: Location, miningLocations: List<Location>): Cell {
+            return when (location in miningLocations) {
+                true -> mine(location)
+                false -> safe(location)
+            }
+        }
```

Comment:
클래스 레이아웃에 대한 컨벤션을 고려한다면 `safe`, `mine`은 `of` 아래로 위치하는것이 더 자연스러울것 같습니다~

https://kotlinlang.org/docs/coding-conventions.html#class-layout
> Do not sort the method declarations alphabetically or by visibility, and do not separate regular methods from extension methods. Instead, put related stuff together, so that someone reading the class from top to bottom can follow the logic of what's happening. Choose an order (either higher-level stuff first, or vice versa) and stick to it.